src/ape/api/providers.py

```diff
@@ -23,24 +23,40 @@ class TransactionAPI:
     nonce: Optional[int] = None  # NOTE: `Optional` only to denote using default behavior
     value: int = 0
     gas_limit: Optional[int] = None  # NOTE: `Optional` only to denote using default behavior
-    gas_price: Optional[int] = None  # NOTE: `Optional` only to denote using default behavior
     data: bytes = b""
+    type: str = ""
```

Comment:
Can type be an enum?

Reply:
I think maybe the API property should be a native type, like `str`, so it is most extensible? Otherwise, there is no good way to have like a "base" enum. `ape-ethereum` will use the `0x0` and `0x2` but not every ecosystem may use those same transaction versions (some may exclude `0x2` altogether for example).

Reply:
I can use enum in `ape-ethereum` at the very least though, to validate

Reply:
`TransactionStatusEnum` is similar, except less of a big deal because int-enums can be argued as just being arbitrarily ordered.